modules/navier_stokes/include/physics/PNSFVSolidHeatTransferPhysics.h

```diff
@@ -0,0 +1,73 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "HeatConductionFV.h"
+#include "NS.h"
+
+/**
+ * Creates all the objects needed to solve the porous media solid energy equation
+ */
+class PNSFVSolidHeatTransferPhysics final : public HeatConductionFV
+{
+public:
+  static InputParameters validParams();
+
+  PNSFVSolidHeatTransferPhysics(const InputParameters & parameters);
+
+protected:
+private:
+  void addNonlinearVariables() override;
+  void addFVKernels() override;
+  void addMaterials() override;
+
+  // Note that we inherit:
+  // - addInitialConditions from HeatConductionPhysics
+  // - addPreconditioning from HeatConductionPhysics
+  // - addFVBCs from HeatConductionFV
+
+  InputParameters getAdditionalRMParams() const override;
+
+  /**
+   * Functions adding kernels for the solid energy equation
+   */
+  void addPINSSolidEnergyTimeKernels();
+  void addPINSSolidEnergyHeatConductionKernels();
+  void addPINSSolidEnergyAmbientConvection();
+  void addPINSSolidEnergyExternalHeatSource();
+
+  // TODO Add interfaces
+
+  /// Process thermal conductivity (multiple functor input options are available).
+  /// Return true if we have vector thermal conductivity and false if scalar
+  bool processThermalConductivity();
+
+  /// Solid temperature name
+  const NonlinearVariableName _solid_temperature_name;
+  /// Fluid temperature name
+  const NonlinearVariableName _fluid_temperature_name;
+  /// Name of the porosity functor (usually material property)
+  const MooseFunctorName _porosity_functor_name;
+  /// Name of the density functor (usually material property)
+  const MooseFunctorName _density_name;
+  /// Name of the specific heat functor (usually material property)
+  const MooseFunctorName _specific_heat_name;
```

Comment:
Let's be consistent about having "functor" in the variable name or not

Reply:
removed functor